### DISCUSSION TITLE
Access Async-Storage data from the native container

### DISCUSSION THREAD

**Discussion by awaldron** [react-native-async-storage/async-storage#539] [Q&A]
## You want to:  
Ask a question

## Details:
I'm building an app that will require some native functionality. I'm storing the auth token in AsyncStorage and I'd like the native app to be able to access that token. Is that possible?

**Comment by krizzu** [comment]
Currently it's only available [for iOS only, through delegate](https://react-native-async-storage.github.io/async-storage/docs/advanced/brownfield)

**Reply by axeljeremy7** [reply]
how is the import for AppDelegate.m  since I have error  bridge not found when I try to use it.

**Reply by umang-simform** [reply]
@Krizzu I am also confused about how to use it?

**Reply by krizzu** [reply]
Would this [small example usage](https://github.com/react-native-async-storage/async-storage/blob/master/example/ios/AsyncStorageExample/AsyncStorageDevSupport.m#L62) be helpful?

**Reply by umang-simform** [reply]
@Krizzu  Thank you for the reply, I already have these 2 files added to our project, I am just confused about how to utilize this?
 So any example of fetching all keys from the storage would be really helpful, Like how to import file in obj-c let's say we want to access data inside appDelegate then how can we do that ?

**Reply by axeljeremy7** [reply]
@Krizzu I don't know objective c but when I add those files I get error like class not found  or the interface methods are missing , and I think is. that I dont know how to added  and I am still unable to use it. Has someone has done it? the example is not even a react native project. My idea is to add a shared instance to appDelegate where I have function for notifications so based on the storage I can show or not the remote notification.
